[marchjlim] iP

marchjlim opened the pull request:
**Phenex**
>"Aiken Dueet" - Tan Tuck Choy, Aaron
Phenex frees your mind of having to remember things you need to do. It's,
- Easy to learn
- Text based
- ~~FAST~~ SUPER FAST to use

All you need to do is
1. Download it [here](https://github.com/marchjlim/ip)
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE!**

Features:
- [x] Managing tasks
- [x] Managing deadlines
- [ ] Reminders (Coming soon!)

If you are a Java programmer, you can use it to practice Java too. Here's the main method:
```
public static void main(String[] args) {
        String home = System.getProperty("user.home");
        Path filePath = Paths.get(home, "Downloads", "CS2103T_AY2425", "iP", "data", "phenex.txt");
        Phenex phenex = new Phenex(filePath);

        phenex.ui.greet();

        Scanner scanner = new Scanner(System.in);
        String userInput;
        Command command = null;

        while (scanner.hasNext()) {
            // scan inputs
            userInput = scanner.nextLine();

            phenex.ui.printLine();

            try {
                command = phenex.parser.parseCommandFromLine(userInput);
                command.execute(phenex.tasks, phenex.ui, phenex.storage);
            } catch (PhenexException e) {
                Ui.printExceptionMessage(e);
            }

            phenex.ui.printLine();
            
            if (command instanceof TerminatingCommand) {
                break;
            }
        }

        scanner.close();
    }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

marchjlim committed:
Renamed Duke to Phenex
Added Greeting/Farewell messages

marchjlim committed:
Updated Phenex messages

marchjlim committed:
Added echo feature for Phenex

marchjlim committed:
Add, and List feature added to Phenex

marchjlim committed:
Added Task class to encapsulate a Task

marchjlim committed:
Changed Task.java toString method. Implemented getStatusIcon method.

Updated Phenex to contain regex support. Added regex features to better detect terminating, and list string commands.
Added mark task as done, or undone features for phenex.

marchjlim committed:
changed some methods and some fields to protected in Task to protected.
Added Deadline, Event, and ToDo classes

marchjlim committed:
Updated Phenex to be an Object. Added instance methods for Phenex.

marchjlim committed:
added ToDo, Deadline, Event functionality to Phenex

marchjlim committed:
Added printTaskAdded method to allow for code reusability

marchjlim committed:
Added test cases for input.txt and EXPECTED.txt

marchjlim committed:
Added TaskType enum. Added addTask method to Phenex, and used switch statements for code reusability.

marchjlim committed:
Added Phenex Exception Class

marchjlim committed:
Added PhenexException error handling

marchjlim committed:
Added delete functionality to Phenex

marchjlim committed:
Add style changes

marchjlim committed:
Merge branch 'branch-Style-Fix'

marchjlim committed:
Add save feature for Phenex

marchjlim committed:
Merge branch 'branch-Level-7'

marchjlim committed:
add LocalDate support for Phenex

marchjlim committed:
Merge branch 'branch-Level-8'

marchjlim committed:
Add Storage, TaskList, Ui classes

marchjlim committed:
Remove unused imports and allowed Phenex and Storage to take in Path in their constructors

marchjlim committed:
Add more OOP classes: TaskList, Ui, Parser, Storage

marchjlim committed:
Merge branch 'branch-A-MoreOOP'

marchjlim committed:
Add packages

marchjlim committed:
Merge branch 'branch-A-Packages'

marchjlim committed:
Updated gitignore

marchjlim committed:
Updated gitignore file to ignore phenex.txt

marchjlim committed:
Merge branch 'add-gradle-support'

marchjlim committed:
Add JavaDoc comments to all files

marchjlim committed:
Add coding standard changes

marchjlim committed:
Add find feature

marchjlim committed:
Merge branch 'branch-A-CodingStandard'

marchjlim committed:
Resolve merge conflicts

marchjlim committed:
Resolve merge conflicts

marchjlim committed:
Add JUnit test

marchjlim committed:
Merge branch 'branch-A-JUnit'

marchjlim committed:
Add Command class and updated Parser

marchjlim committed:
Add JavaDoc to Command classes and its subclasses

marchjlim committed:
Add JavaDoc and remove unused import, methods, enum

marchjlim committed:
Merge branch 'branch-StyleFix'

marchjlim committed:
Add more JavaDoc

marchjlim committed:
Update README.md

marchjlim committed:
Add checkstyle

marchjlim committed:
Merge branch 'branch-A-CheckStyle'

dominic2412 commented on src/main/java/phenex/storage/Storage.java:
```diff
@@ -0,0 +1,62 @@
+    public Storage(Path filePath) {
+        this.filePath = filePath;
+
+        if (!Files.exists(this.filePath)) {
```
I like how you’ve handled the initialization of the storage file. It’s a good practice to ensure that necessary resources are in place before proceeding.

dominic2412 commented on src/main/java/phenex/storage/Storage.java:
```diff
@@ -0,0 +1,62 @@
+                String line = Parser.parseTaskInfo(task);
+                fileWriter.write(line);
+            }
+            fileWriter.close();
```
You might want to consider using a try-with-resources statement here to automatically close FileWriter and avoid potential resource leaks.

dominic2412 commented on src/test/java/phenex/task/TaskListTest.java:
```diff
@@ -0,0 +1,43 @@
+    public void invalid_index_phenexExceptionThrown() {
+        TaskList taskList = new TaskList();
+        ToDo toDoTask = new ToDo("task Test");
+        taskList.addTask(toDoTask);
```
This part of the code is quite clear and easy to follow. I appreciate the effort to maintain readability while performing multiple operations.

dominic2412 reviewed (commented):
Overall, your implementation is solid, and I especially appreciate how you've structured the code into logical, cohesive classes. There are just a few areas where we could potentially enhance clarity or functionality, but those are minor adjustments. Great job!

marchjlim committed:
Add GUI functionalities to Phenex

marchjlim committed:
Refactor Phenex GUI to use FXML

marchjlim committed:
Cleaned style of code to adhere to Java Coding Standard

marchjlim committed:
Merge branch 'branch-Level-10'

marchjlim committed:
Refactor code to use Varargs

marchjlim committed:
Merge branch 'branch-A-Varargs'

marchjlim committed:
Refactor code to adhere to Java Coding Standard

marchjlim committed:
Add further functionality for Phenex GUI

marchjlim committed:
Merge branch 'branch-improve-gui'

muller317 commented on src/main/java/phenex/task/Deadline.java:
```diff
@@ -0,0 +1,45 @@
+    }
+
+    @Override
+    public String formatDate(LocalDate localDate) {
```
Would it be better to name it as getformatDate?

muller317 commented on src/main/java/phenex/task/Deadline.java:
```diff
@@ -0,0 +1,45 @@
+    }
+
+    @Override
+    public boolean overlapsWith(LocalDate localDate) {
```
Would it be better to name it as isOverlapsWith?

muller317 reviewed (commented):
Overall the coding standards are well adhered to.

marchjlim committed:
Add assert for TaskList class

There are several methods which manipulate the TaskList and its
contents depending on the method parameters. As such, it is
crucial that the parameters are not invalid, as this logic is
handled by the programmer.

Including asserts will help guard against programmer's errors of
passing in invalid parameters to these methods.

marchjlim committed:
Refactor various methods in Ui class

There are several methods which print a string and then return it.
This violates the SLAP principle, so methods which generate a string
have been created to refactor the code and adhere to SLAP principle.

A step towards adhering to better code quality.

marchjlim committed:
Refactor Phenex class to improve code quality

marchjlim committed:
Refactor various methods in Parser class

There are several methods have different levels of abstraction.
This violates the SLAP principle, so more methods
have been created to refactor the code and adhere to SLAP principle.

A step towards adhering to better code quality.

marchjlim committed:
Refactor TaskList methods to add stream functionality

marchjlim committed:
Add minor change to findAllTasks method in TaskList class

marchjlim committed:
Merge pull request #3 from marchjlim/branch-A-CodeQuality

Branch a code quality

marchjlim committed:
Merge commit '6e326a9746d64a958b3c62ded0a16485bb130263'

marchjlim committed:
Revert "Merge commit '6e326a9746d64a958b3c62ded0a16485bb130263'"

This reverts commit 515aae4afd09142ede07d66342d70890ace95d7f, reversing
changes made to 16c0a1ebe7342c78fa3a5a5f9e3bff5f54262352.

marchjlim committed:
Merge branch 'master' into branch-A-Streams

# Conflicts:
#	src/main/java/phenex/task/TaskList.java

marchjlim committed:
Merge branch 'master' into branch-A-Assertions

marchjlim committed:
Merge pull request #2 from marchjlim/branch-A-Assertions

Add assert for TaskList class

marchjlim committed:
Merge pull request #1 from marchjlim/branch-A-Streams

Branch a streams

marchjlim committed:
Add equals method to Task, Event, Deadline

marchjlim committed:
Add feature to prevent duplicate tasks from being added

marchjlim committed:
Merge branch 'branch-BCD-Extension'

marchjlim committed:
Resize image dimensions and flip phenex image

marchjlim committed:
Merge branch 'branch-A-BetterGui'

marchjlim committed:
Add new style and personality for Phenex GUI

marchjlim committed:
Merge branch 'branch-A-Personality'

marchjlim committed:
Add error handling for events that start and end on same date

marchjlim committed:
Merge branch 'branch-A-MoreErrorHandling'

marchjlim committed:
Add more TaskList and Deadline unit testing

marchjlim committed:
Add Even, Deadline, and TaskList unit tests with aid of AI

marchjlim committed:
Merge branch 'branch-A-AiAssisted'

marchjlim committed:
Add AI.md

marchjlim committed:
Fix bug in GUI where last line would not be printed

marchjlim committed:
Refactor code to apply SLAP

marchjlim committed:
Merge branch 'branch-finalise-features'

marchjlim committed:
Refactor code to improve GUI

marchjlim committed:
Add user guide for Phenex

marchjlim committed:
Merge branch 'branch-A-UserGuide'

marchjlim committed:
Add more details to User Guide

marchjlim committed:
Add new Ui image

marchjlim committed:
Add GUI resizeable feature

marchjlim committed:
Add feature that ensures relative filepath is used for data

marchjlim committed:
Merge branch 'branch-resize-gui'

marchjlim committed:
Fix invalid date bug and file writing bug

marchjlim committed:
Fix date time bug

marchjlim committed:
Merge branch 'branch-filepath-bugfix'

marchjlim committed:
Fix indexing bug

marchjlim committed:
Add text wrapping for dialog box, application close on bye features

marchjlim committed:
Add schedule feature

marchjlim committed:
Enhance schedule feature

marchjlim committed:
Enhance mark command and Tasks to encapsulate hours

marchjlim committed:
Fix storage bugs with mark command

marchjlim committed:
Fix bugs, update dependencies

marchjlim committed:
Add task type to Task

marchjlim committed:
Fix parser bugs for Todo

marchjlim committed:
Enhance list command by adding more details

marchjlim committed:
Fix bugs

marchjlim committed:
Fix bugs to ensure functionality of find command

marchjlim committed:
Enhance find command to allow optional parameters

marchjlim committed:
Remove unused code

marchjlim committed:
Merge pull request #6 from marchjlim/find-enhancement

Find enhancement
